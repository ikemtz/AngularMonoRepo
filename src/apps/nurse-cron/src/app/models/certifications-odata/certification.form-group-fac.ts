/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export function CertificationFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl<string | null>(null),
    name: new FormControl<string | null>(null),
    isEnabled: new FormControl<boolean | null>(null),
    expiresOnUtc: new FormControl<Date | null>(null),
  });
}
