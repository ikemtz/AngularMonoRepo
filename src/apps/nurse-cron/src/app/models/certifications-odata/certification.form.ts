/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup } from '@angular/forms'; //NOSONAR

export interface ICertificationForm {
  id?: FormControl<string | null>;
  name?: FormControl<string | null>;
  isEnabled?: FormControl<boolean | null>;
  expiresOnUtc?: FormControl<Date | null>;
}
