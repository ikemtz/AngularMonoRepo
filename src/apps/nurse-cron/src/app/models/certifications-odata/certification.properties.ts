/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum CertificationProperties {
  ID = 'id',
  NAME = 'name',
  IS_ENABLED = 'isEnabled',
  EXPIRES_ON_UTC = 'expiresOnUtc',
}
