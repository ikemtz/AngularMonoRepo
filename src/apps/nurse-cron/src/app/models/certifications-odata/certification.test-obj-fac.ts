/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #test-object-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */

import { CertificationProperties } from './certification.properties';

export function createTestCertification() {
    return { 
      [CertificationProperties.ID]: 'ID',
      [CertificationProperties.NAME]: 'NAME',
      [CertificationProperties.IS_ENABLED]: false,
      [CertificationProperties.EXPIRES_ON_UTC]: new Date(), 
    };
}
