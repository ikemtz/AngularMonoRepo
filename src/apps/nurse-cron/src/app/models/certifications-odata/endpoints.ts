/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #endpoints-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum Endpoints {
  certifications = '/odata/v1/Certifications',
}
export interface IEndpoints {
  readonly certifications: string;
}
