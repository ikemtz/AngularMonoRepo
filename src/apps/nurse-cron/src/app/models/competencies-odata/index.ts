/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.form';
export * from './competency.model';
export * from './competency.properties';
export * from './competency.test-obj-fac';
