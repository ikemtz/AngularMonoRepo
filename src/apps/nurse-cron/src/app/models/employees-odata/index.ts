/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './employee-certification.form-group-fac';
export * from './employee-certification.form';
export * from './employee-certification.model';
export * from './employee-certification.properties';
export * from './employee-certification.test-obj-fac';
export * from './employee-competency.form-group-fac';
export * from './employee-competency.form';
export * from './employee-competency.model';
export * from './employee-competency.properties';
export * from './employee-competency.test-obj-fac';
export * from './employee-health-item.form-group-fac';
export * from './employee-health-item.form';
export * from './employee-health-item.model';
export * from './employee-health-item.properties';
export * from './employee-health-item.test-obj-fac';
export * from './employee.form-group-fac';
export * from './employee.form';
export * from './employee.model';
export * from './employee.properties';
export * from './employee.test-obj-fac';
