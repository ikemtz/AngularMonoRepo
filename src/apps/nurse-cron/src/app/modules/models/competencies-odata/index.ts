/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.model';
export * from './competency.properties';
