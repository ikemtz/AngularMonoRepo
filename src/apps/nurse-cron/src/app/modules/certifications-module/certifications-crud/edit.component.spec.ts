import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { ImngDataEntryDialogModule } from 'imng-kendo-data-entry';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { DatePickerModule } from '@progress/kendo-angular-dateinputs';
import { of } from 'rxjs';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';

import { CertificationEditComponent } from './edit.component';
import { CertificationCrudFacade } from './crud.facade';
import { CertificationProperties, ICertification } from '../../../models/certifications-odata';

describe('CertificationEditComponent', () => {
  let component: CertificationEditComponent;
  let fixture: ComponentFixture<CertificationEditComponent>;
  let facade: CertificationCrudFacade;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [CertificationEditComponent],
      imports: [ReactiveFormsModule, ImngDataEntryDialogModule, NoopAnimationsModule, DatePickerModule],
      providers: [
        {
          provide: CertificationCrudFacade,
          useValue: createDataEntryMockFacade({ currentEntity$: of({}) }),
        },
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(CertificationEditComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(CertificationCrudFacade);
    fixture.detectChanges();
  });

  it('should update', () => {
    component.initForm();
    component.addEditForm.patchValue({
      [CertificationProperties.ID]: 'ID',
      [CertificationProperties.NAME]: 'NAME',
      [CertificationProperties.IS_ENABLED]: true,
      [CertificationProperties.EXPIRES_ON_UTC]: new Date(),
    });
    let item: ICertification;
    facade.updateExistingEntity = jest.fn(x => (item = x));
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(1);

    expect(item).toMatchSnapshot({
      expiresOnUtc: expect.any(Date),
    });

  });

  it('should not update', () => {
    component.addEditForm = { valid: false } as never;
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);
  });

  it('should cancel', () => {
    component.cancel();
    expect(facade.clearCurrentEntity).toBeCalledTimes(1);
  });
});
