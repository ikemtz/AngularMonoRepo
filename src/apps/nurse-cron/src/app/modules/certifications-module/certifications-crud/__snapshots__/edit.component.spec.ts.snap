// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificationEditComponent should update 1`] = `
{
  "expiresOnUtc": Any<Date>,
  "id": "ID",
  "isEnabled": true,
  "name": "NAME",
}
`;
